automation/planner.py

```diff
@@ -19,28 +18,27 @@
 def get_command(name: str, range_: DateTimeRange):
     return RECEIVER_COMMAND + '"%s" "%s"' % (name, range_.end_datetime.isoformat())
 
-def get_passes(config, from_: datetime.datetime, to: datetime.datetime):
-    location = Location(config["location"]["name"],
-        config["location"]["latitude"], config["location"]["longitude"],
-        config["location"]["elevation"])
+def get_passes(config: Configuration, from_: datetime.datetime, to: datetime.datetime):
+    location = Location(*get_location(config))
     satellties = filter(lambda s: not s.get("disabled", False), config["satellites"])
-    strategy_name = config.get("strategy", "max-elevation")
+    strategy_name: str = config.get("strategy", "max-elevation") # type: ignore
 
     orbit_db = OrbitDatabase(config["norad"])
     strategy = strategy_factory(strategy_name)
     
     init = []
     for sat in satellties:
-        aos_at = sat.get("aos_at") or config.get("aos_at") or 0
-        max_elevation_greater_than = sat.get("max_elevation_greater_than") or config.get("max_elevation_greater_than") or 0
+        set_satellite_defaults(config, sat)
+        aos_at = sat["aos_at"]
+        max_elevation_greater_than = sat["max_elevation_greater_than"]
         predictor = orbit_db.get_predictor(sat["name"])
         passes = predictor.passes_over(location, from_, to, max_elevation_greater_than, aos_at_dg=aos_at)
         init += [(sat["name"], p) for p in passes]
     
     selected = strategy(init)
     return selected
 
-def plan_passes(selected: Iterable[Observation], cron):
+def plan_passes(selected: Sequence[Observation], cron):
     selected = sorted(selected, key=lambda o: o.pass_.aos)
     for entry in selected:
         cmd = get_command(entry.data, entry.range)
```

Comment:
Is user expected to ever run planner.py? There's no help and no explanation what kind of parameters are expected. Had to look in the source code.

Reply:
The possibility to run this script directly was useful for debug. I never need to run it this way in production.

Reply:
This script is running by CRON. User should execute it as script. Therefore it use ```sys.argv``` but no ```argparse```. ```pprint``` call isn't necessary.